### DISCUSSION TITLE
Why can I see the contents of files in the attachment field even when I'm logged out via the URL link to the file?

### DISCUSSION THREAD

**Discussion by neojigi** [Budibase/budibase#13782] [Help] [answered]
my env

{
  "browser": {
    "language": "ko-KR",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0",
    "platform": "Win32",
    "vendor": ""
  },
  "server": {
    "budibaseVersion": "2.27.3",
    "hosting": "docker",
    "nodeVersion": "v20.11.1",
    "platform": "linux",
    "cpuArch": "x64",
    "cpuCores": 4,
    "cpuInfo": "Intel(R) N97",
    "totalMemory": "15.371818542480469GB",
    "uptime": "2 day(s), 1 hour(s), 52 minute(s)"
  }
}

I am using internal db.

When I upload a picture file in the attachment field, I can copy the URL link to the picture file.
When I type that link into the address bar of my browser and hit enter, the file is displayed even when I am logged out!
I think this is a very big security issue.

I've seen discussions around this topic in 2021.
The person responding at the time said it was a bug and that it would be fixed soon, but I don't think it's been addressed yet.

I want to make that link available only to logged-in users, but how can I do that?
If I enter through the minio browser, is there anything I can change there?

**Comment by aptkingston** [comment, marked as answer]
Hey @neojigi. The URLs for attachments are generated on the fly by the server, and are pre-signed. That means that the URL does give you access to the attachment regardless of whether you're logged in.

However, the important thing to note is that the attachment URLs are ephemeral - they only last for one hour, at which point they'll stop working. You won't be able to generate a new one unless you're logged in, so the only way that logged out users can access attachments is if you log in, generate a URL and share it with a logged out user, which gives you a 1 hour window until the link stops working.

We have made significant improvements here since those 2021 discussions. The URLs used to work forever, which absolutely was an issue. The signed URL updates we did were specifically to address this issue.

Also just going to convert this to a discussion and we can continue the conversation there if you have any more questions.
